lib/internal/flutter_ble_lib_mixins/discovery_mixin.dart

```diff
@@ -0,0 +1,60 @@
+part of flutter_ble_lib;
+
+mixin DiscoveryMixin on FlutterBLE {
+  Future<void> discoverAllServicesAndCharacteristics(
+      Peripheral peripheral, String transactionId) async {
+    await _methodChannel.invokeMethod(
+        MethodName.discoverAllServicesAndCharacteristics, <String, dynamic>{
+      ArgumentName.deviceIdentifier: peripheral.identifier,
+      ArgumentName.transactionId: transactionId
+    });
+  }
+
+  Future<List<Service>> services(Peripheral peripheral) async {
+    String jsonString = await _methodChannel.invokeMethod(
+        MethodName.services, <String, dynamic>{
+      ArgumentName.deviceIdentifier: peripheral.identifier
+    });
+
+    List<String> decodedJson = (jsonDecode(jsonString) as List<dynamic>).cast();
+
+    return decodedJson
+        .map((serviceJsonString) =>
+            Service.fromJson(jsonDecode(serviceJsonString), peripheral, _manager))
+        .toList();
+  }
+
+  Future<List<Characteristic>> characteristics(
+      Peripheral peripheral, String serviceUuid) async {
+    String jsonString = await _methodChannel
+        .invokeMethod(MethodName.characteristics, <String, dynamic>{
+      ArgumentName.deviceIdentifier: peripheral.identifier,
+      ArgumentName.serviceUuid: serviceUuid
+    });
```

Comment:
It will allow as to reformat it in a nice way if we would a comma before the last ")" or before "}"